fineract-provider/src/main/resources/db/changelog/tenant/parts/0012_pentaho_reports_to_table.xml

```diff
@@ -0,0 +1,247 @@
+            <column name="report_category" value="Accounting"/>
+            <column name="report_sql" value="select * from (select debits.glcode as &quot;glcode&quot;,  debits.name as &quot;name&quot;, IF(debits.type = 1 or debits.type = 5,  ifnull(debits.debitamount,0)-ifnull(credits.creditamount,0),null) as &quot;debit&quot;,  IF(debits.type = 4 or debits.type = 3 or debits.type = 2,  ifnull(credits.creditamount,0)-ifnull(debits.debitamount,0),null) as &quot;credit&quot; from (select acc_gl_account.gl_code as &quot;glcode&quot;, name,sum(amount) as &quot;debitamount&quot;, acc_gl_account.classification_enum as &quot;type&quot; from acc_gl_journal_entry,acc_gl_account where acc_gl_account.id = acc_gl_journal_entry.account_id and acc_gl_journal_entry.type_enum=2 and acc_gl_journal_entry.entry_date between date('${startDate}') and date('${endDate}') and (acc_gl_journal_entry.office_id= ${officeId} or ${officeId}=1) group by glcode order by glcode) debits LEFT OUTER JOIN (select acc_gl_account.gl_code as &quot;glcode&quot;, name as &quot;name&quot;, sum(amount) as &quot;creditamount&quot;, acc_gl_account.classification_enum as &quot;type&quot; from acc_gl_journal_entry,acc_gl_account where acc_gl_account.id = acc_gl_journal_entry.account_id and acc_gl_journal_entry.type_enum=1 and acc_gl_journal_entry.entry_date between date('${startDate}') and date('${endDate}') and (acc_gl_journal_entry.office_id= ${officeId} or ${officeId}=1) group by glcode order by glcode) credits on debits.glcode=credits.glcode union select credits.glcode as &quot;glcode&quot;,  credits.name as &quot;name&quot;, IF(credits.type = 1 or credits.type = 5,  ifnull(debits.debitamount,0)-ifnull(credits.creditamount,0),null) as &quot;debit&quot;,  IF(credits.type = 4 or credits.type = 3 or credits.type = 2,  ifnull(credits.creditamount,0)-ifnull(debits.debitamount,0),null) as &quot;credit&quot; from (select acc_gl_account.gl_code as &quot;glcode&quot;, name, sum(amount) as &quot;debitamount&quot;, acc_gl_account.classification_enum as &quot;type&quot; from acc_gl_journal_entry,acc_gl_account where acc_gl_account.id = acc_gl_journal_entry.account_id and acc_gl_journal_entry.type_enum=2 and acc_gl_journal_entry.entry_date between date('${startDate}') and date('${endDate}') and (acc_gl_journal_entry.office_id= ${officeId} or ${officeId}=1) group by glcode order by glcode) debits RIGHT OUTER JOIN (select acc_gl_account.gl_code as &quot;glcode&quot;,name as &quot;name&quot;, sum(amount) as &quot;creditamount&quot;, acc_gl_account.classification_enum as &quot;type&quot; from acc_gl_journal_entry,acc_gl_account where acc_gl_account.id = acc_gl_journal_entry.account_id and acc_gl_journal_entry.type_enum=1 and acc_gl_journal_entry.entry_date between date('${startDate}') and date('${endDate}') and (acc_gl_journal_entry.office_id= ${officeId} or ${officeId}=1) group by glcode order by glcode) credits on debits.glcode=credits.glcode) as fullouterjoinresult order by glcode "/>
```

Comment:
The query clearly won't work on PostgreSQL since it's using the IFNULL function which is not supported by PostgreSQL. Instead we should use COALESCE but I assume there are other problems as well with the query.